src/binder/utils.cpp

```diff
@@ -300,7 +303,9 @@ namespace BINDER_SPACE
         BINDER_LOG_STRING(W("path A"), pathA);
         BINDER_LOG_STRING(W("path B"), pathB);
 
-        WCHAR tempResultPath[MAX_LONGPATH];
+        PathString tempResultPathString;
+        WCHAR * tempResultPath = tempResultPathString.OpenUnicodeBuffer(MAX_LONGPATH);
+
         if (PathCombineW(tempResultPath, pathA, pathB))
```

Comment:
The contract for `PathCombine`  is hardcoded to `MAX_PATH`. The right fix here is to implement a simple path combining here, without calling PathCombine:
- Copy `pathA` to `combinedPath`
- Append path separator to `combinedPath` if it does not end with one
- Append `pathB` to `combinedPath`